Merge node sections on production deployment page (#832)
redpanda-data/documentation: docs/platform/deployment/production-deployment.mdx

:::

## Configure and start nodes

Configure and start up the nodes with the [`rpk redpanda config bootstrap`](../../reference/rpk/rpk-redpanda/rpk-redpanda-config-bootstrap) command: 

```bash
sudo rpk redpanda config bootstrap --self <private ip> --ips <seed nodes ips> && \
sudo rpk redpanda config set redpanda.empty_seed_starts_cluster false && \
sudo systemctl start redpanda-tuner redpanda
```

- The `--self` flag tells the node which interface address to bind to. Usually this is its private IP.

> - The `--ips` flag lists the all seed nodes in the cluster (equivalent to the `seed_servers` property in `redpanda.yaml`).

Reviewer: Change "the all" to "all the"
```suggestion
- The `--ips` flag lists all the seed nodes in the cluster (equivalent to the `seed_servers` property in `redpanda.yaml`).
```

> A Redpanda cluster starts with the seed nodes that are specified in the `--ips` flag, instantiating a controller Raft group with all the seed nodes at once. The cluster is then available after all seed nodes complete their startup procedure and become accessible. After that, the cluster's node count grows as non-seed nodes start up.

Reviewer: See if this is more clear:

```suggestion
When a Redpanda cluster starts, it instantiates a controller Raft group with all the seed nodes that are specified in the `--ips` flag. After all seed nodes complete their startup procedure and become accessible, the cluster is then available . After that, non-seed nodes start up and are added to the cluster.
```

```

- The `--self` flag tells the node which interface address to bind to. Usually this is its private IP.

> - The `--ips` flag lists the all seed nodes in the cluster (that is, seed servers).

Reviewer 2: nit: maybe the `(that is, seed servers)` parenthesis is not needed -- there is no new information in them to my taste

Author: I removed the old wording and then added how the `ips` flag is related to `seed_servers`.

> - The `--ips` flag lists the all nodes in the cluster (ie. seed servers). This must be identical on all nodes when deploying a cluster.

Reviewer 2: Actually `--ips` should only list seed nodes, not all nodes in the cluster. There is a discussion of roles of seed nodes [in the TOI document](https://vectorizedio.atlassian.net/wiki/spaces/CORE/pages/273907801/Homogenous+Node+Config+And+Improved+Cluster+Bootstrap#Seeds-Driven-Cluster-Bootstrap.1). Note that the `--ips` argument actually feeds the `seed_servers` property, so everything related to that property in the doc is also applicable to `--ips`.

Author: This makes sense. I assumed that users would want to have all nodes serve as seed servers. If not, then what are the scenarios where users wouldn't want to do this? It ends up being a difference in nodes that worries me if we distinguish it in this way, especially when considering our arguments against KRaft.

Reviewer 2: Right, at the moment of cluster bootstrap those most probably are just all nodes, however that is not required, and I just wanted to avoid creating an impression about the otherwise, esp. in the "Deployment for Production" section which seems to be an advanced one. The other scenario may be that the customer wants really many nodes in the cluster, but not all of them as seed nodes to optimize time for cluster to become ready to use. So maybe we can at least change the strong wording like `This must be identical on all nodes...` since technically that is not quite correct, and also make the assumption you're talking about clear to the reader.

> - The `--ips` flag lists the all nodes in the cluster (ie. seed servers). This must be identical on all nodes when deploying a cluster.

Reviewer 3: ```suggestion
- The `--ips` flag lists the all nodes in the cluster (that is, seed servers). This must be identical on all nodes when deploying a cluster.
```

> A Redpanda cluster starts with the seed nodes that are specified in the `--ips` flag, instantiating a controller Raft group with the seed nodes. The cluster's node count grows as nodes start up. The cluster is then available after all seed servers complete their startup procedure and become accessible.

Reviewer 2: I would rearrange that a bit. 
```suggestion
A Redpanda cluster starts with the seed nodes that are specified in the `--ips` flag, instantiating a controller Raft group with all the seed nodes at once. The cluster is then available after all seed nodes complete their startup procedure and become accessible. After that, the cluster's node count grows as non-seed nodes start up.
```
The idea is like:
* all seed nodes form the very initial cluster
* cluster becomes availalable to clients
* non-seed nodes join and the cluster grows

Note that if we _restart_ an existing cluster, the sentnece about "cluster availble when all seed nodes completes startup" is not applicable anymore, the cluster becomes available once its consensus is able to elect a leader, i.e. once at least a majority of all current cluster members complete startup.

Author: Thanks for the explanations @dlex ! I've updated this PR to reflect the changes.
